### DISCUSSION TITLE
New REST API client: Bruno

### DISCUSSION THREAD

**Discussion by ntarocco** [inveniosoftware/docs-invenio-rdm#635] [Trainings Completed]
Speakers: @ntarocco 
Description: demonstrate the recent REST API client [Bruno](https://usebruno.com) as a team tool for Invenio/InvenioRDM/InvenioILS.

**Comment by ntarocco** [comment]
* When: Tue 16 Apr 2024
* Duration: 20 minutes demo - 10 minutes discussion

The Bruno REST API client (Oct 2023) was introduced, highlighting its potential advantages for our teams and projects. It provides a Desktop app, a CLI (compatible with CI), and a VsCode extension. It offers a practical alternative to tools such as Postman or Insomnia.

The main benefit is that it's both free and open-source. Plus, it automatically saves all the collections/endpoints you define via the UI in a file (or set of files) using a simple markup language called `Bru`. These files can then be versioned in a GIT repository. This means our teams can all use the same tool and GIT repo, making it easy to gradually build and maintain REST API integrations collaboratively. The open source community of our projects can also benefit from this.

Other notable features include the ability to create small scripts for manipulating requests/responses (e.g., injecting fake data into the body), support for assertions/tests that can be executed in a CI environment, and support for environment variables and secrets.

In the [roadmap](https://github.com/usebruno/bruno/discussions/384), there are upcoming features in v3 and v4 that look interesting. For instance, the automatic generation of documentation from the REST API could be really helpful for us in keeping the InvenioRDM documentation up-to-date.

The project appears to be sustainable, with a commitment to remaining open and free. See the related discussion [here](https://github.com/usebruno/bruno/discussions/269).
Some other useful links:
* [Bruno: First thoughts](https://dev.to/kleeut/bruno-first-thoughts-5nc)
* [bruno-user-guide](https://github.com/Automation-Test-Starter/Bruno-API-Test-Starter): a collection of useful examples.

**Comment by slint** [comment]
I've created a GitHub repo and added 2 collections for creating a minimal record and retrieving records via the InvenioRDM REST API: https://github.com/slint/inveniordm-rest-api-collection
